File: _algorithms_challenges/projecteuler/ProjectEuler-master(2)/ProjectEuler-master/1/1.pl
use strict;
use warnings;

sub sum_of_multiple {
    my ($multiple, $bound) = @_;
    my $multiple_bound = $bound - ($bound % $multiple);
    return $multiple_bound * (int($multiple_bound / $multiple) + 1) / 2;
}

sub sum_of_multiples {
    my ($bound) = @_;
    my $sum = sum_of_multiple(3, $bound - 1) + sum_of_multiple(5, $bound - 1) 
            - sum_of_multiple(15, $bound - 1);
    print "$sum\n";
}

sub main() {
    sum_of_multiples(10);
    sum_of_multiples(1000);
}

main();

File: _algorithms_challenges/projecteuler/ProjectEuler-master(2)/ProjectEuler-master/1/1.jl
function sum_of_multiple(multiple, bound)
    multiple_bound = bound - (bound % multiple)
    return multiple_bound * (multiple_bound // multiple + 1) // 2
end

function get(bound)
    included = bound - 1
    return sum_of_multiple(3, included) + sum_of_multiple(5, included) - sum_of_multiple(15, included)
end

function main()
    println(get(10))
    println(get(1000))
end

main()
